Source/DafnyLanguageServer/Workspace/Compilation.cs

```diff
@@ -0,0 +1,461 @@
+      return false;
+    }
+
+    var containingModule = canVerify.ContainingModule;
+    if (!containingModule.ShouldVerify(resolution.ResolvedProgram.Compilation)) {
+      return false;
+    }
+
+    if (!onlyPrepareVerificationForGutterTests && !verifyingOrVerifiedSymbols.TryAdd(canVerify, Unit.Default)) {
+      return false;
+    }
+    updates.OnNext(new ScheduledVerification(canVerify));
+
+    if (onlyPrepareVerificationForGutterTests) {
+      await VerifyUnverifiedSymbol(onlyPrepareVerificationForGutterTests, canVerify, resolution);
+      return true;
+    }
+
+    _ = VerifyUnverifiedSymbol(onlyPrepareVerificationForGutterTests, canVerify, resolution);
+    return true;
+  }
+
+  private async Task VerifyUnverifiedSymbol(bool onlyPrepareVerificationForGutterTests, ICanVerify canVerify,
+    Resolution resolution) {
+    try {
+
+      var ticket = verificationTickets.Dequeue(CancellationToken.None);
+      var containingModule = canVerify.ContainingModule;
+
+      IncrementJobs();
+
+      IReadOnlyDictionary<FilePosition, IReadOnlyList<IImplementationTask>> tasksForModule;
+      try {
+        tasksForModule = await translatedModules.GetOrAdd(containingModule, async () => {
+          var result = await verifier.GetVerificationTasksAsync(boogieEngine, resolution, containingModule,
+            cancellationSource.Token);
+          foreach (var task in result) {
+            cancellationSource.Token.Register(task.Cancel);
+          }
+
+          return result.GroupBy(t => ((IToken)t.Implementation.tok).GetFilePosition()).ToDictionary(
+            g => g.Key,
+            g => (IReadOnlyList<IImplementationTask>)g.ToList());
+        });
+      } catch (OperationCanceledException) {
+        verificationCompleted.TrySetCanceled();
+        throw;
+      } catch (Exception e) {
+        verificationCompleted.TrySetException(e);
+        updates.OnNext(new InternalCompilationException(e));
+        throw;
+      }
+
+      // For updated to be reliable, ImplementationsPerVerifiable must be Lazy
+      var updated = false;
+      var implementations = implementationsPerVerifiable.GetOrAdd(canVerify, () => {
+        var tasksForVerifiable =
+          tasksForModule.GetValueOrDefault(canVerify.NameToken.GetFilePosition()) ??
+          new List<IImplementationTask>(0);
+
+        updated = true;
+        return tasksForVerifiable.ToDictionary(
+          t => GetImplementationName(t.Implementation),
+          t => new ImplementationState(t, PublishedVerificationStatus.Stale, Array.Empty<DafnyDiagnostic>(), false));
+      });
+      if (updated) {
+        updates.OnNext(new CanVerifyPartsIdentified(canVerify,
+          implementationsPerVerifiable[canVerify].Values.Select(s => s.Task).ToList()));
+      }
+
+      // When multiple calls to VerifyUnverifiedSymbol are made, the order in which they pass this await matches the call order.
+      await ticket;
+
+      if (!onlyPrepareVerificationForGutterTests) {
```

Comment:
Switch condition to reduce nesting.

Reply:
I don't see how it's possible to reduce nesting here by inverting, but I'll extract something to a separate method.

Reply:
I was thinking of
```cs
if (onlyPrepareVerificationForGutterTests) {
  DecrementJobs();
  return;
}
<previously indented cod>
DecrementJobs();
```
but refactoring is a good idea as well.